tests/cpp/test_multidevice_pipeline.cpp

```diff
@@ -576,6 +581,304 @@ INSTANTIATE_TEST_SUITE_P(
         SchedulingMode::Manual,
         SchedulingMode::ReductionOnly,
         SchedulingMode::Automatic));
+
+class DistributedMatmul : public MultiDeviceTest {
```

Comment:
is there a reason not to use `PipelineTest` as base class? We could reuse the data members and the method executeAndValidate

Reply:
The matmul tests need some extra constraints, some of which are temporary, like disabling the allocation presegmentation pass, ampere architecture or newer, increased validation tolerance, etc.